Source/MQTTnet/Server/Internal/MqttClientConnection.cs

```diff
@@ -282,34 +280,32 @@ async Task SendPacketsLoop(CancellationToken cancellationToken)
                     _logger.Verbose("Client '{0}': Queued application message sent.", ClientId);
                 }
             }
-            catch (OperationCanceledException)
-            {
-            }
             catch (Exception exception)
             {
-                if (exception is MqttCommunicationTimedOutException)
+                if ((queuedApplicationMessage != null) && (queuedApplicationMessage.SubscriptionQualityOfServiceLevel > MqttQualityOfServiceLevel.AtMostOnce))
                 {
-                    _logger.Warning(exception, "Client '{0}': Sending publish packet failed: Timeout.", ClientId);
-                }
-                else if (exception is MqttCommunicationException)
-                {
-                    _logger.Warning(exception, "Client '{0}': Sending publish packet failed: Communication exception.", ClientId);
-                }
-                else
-                {
-                    _logger.Error(exception, "Client '{0}': Sending publish packet failed.", ClientId);
+                    // TODO: losing the order here, need to put back in front of queue
```

Comment:
Please fix the TODO.

Reply:
This TODO was not created by any code in the PR, but it does cite the fact that the queuing order is altered when a message is requeued (as it was before the PR).